esp/esdllib/esdl_script.cpp

```diff
@@ -0,0 +1,917 @@
+                m_prefix.set("xsdl:");
+            else
+            {
+                const char *colon = strchr(tag, ':');
+                if (colon)
+                {
+                    if (colon == tag)
+                        throw MakeStringException(ESDL_SCRIPT_Error, "Tag shouldn't start with colon %s", tag);
+                    m_prefix.append(colon-tag, tag);
+                    if (!streq(m_prefix, "xsdl"))
+                    {
+                        //any prefix other than xsdl must be declared on the root element. in future can support multiple and inherited prefixes
+                        VStringBuffer attpath("@xmlns:%s", m_prefix.str());
+                        const char *uri = tree.queryProp(attpath.str());
+                        if (!uri || !streq(uri, "urn:hpcc:esdl:script"))
+                            throw MakeStringException(ESDL_SCRIPT_Error, "Undeclared script xmlns prefix %s", m_prefix.str());
+                    }
+                    //add back the colon for easy comparison
+                    m_prefix.append(':');
+                }
+            }
+        }
+
+        m_name.set(tree.queryProp("@name"));
+        m_target.set(tree.queryProp("@target"));
+
+        DBGLOG("Compiling custom ESDL Transform: '%s'", m_name.str());
+
+        StringBuffer xpath;
+        Owned<IPropertyTreeIterator> parameters = tree.getElements(makeOperationTagName(xpath, m_prefix, "param"));
+        ForEach(*parameters)
+            m_variables.append(*new CEsdlTransformOperationParameter(&parameters->query(), m_prefix));
+
+        Owned<IPropertyTreeIterator> variables = tree.getElements(makeOperationTagName(xpath.clear(), m_prefix, "variable"));
+        ForEach(*variables)
+            m_variables.append(*new CEsdlTransformOperationVariable(&variables->query(), m_prefix));
+
+        Owned<IPropertyTreeIterator> children = tree.getElements("*");
+        ForEach(*children)
+        {
+            Owned<IEsdlTransformOperation> operation = createEsdlTransformOperation(&children->query(), m_prefix);
+            if (operation)
+                m_operations.append(*operation.getClear());
+        }
+    }
+
+    virtual void appendEsdlURIPrefixes(StringArray &prefixes) override
+    {
+        if (m_prefix.length())
+        {
+            StringAttr copy(m_prefix.str(), m_prefix.length()-1); //remove the colon
+            prefixes.appendUniq(copy.str());
+        }
+        else
+            prefixes.appendUniq("");
+    }
+    virtual void toDBGLog() override
+    {
+#if defined(_DEBUG)
+        DBGLOG(">>>>>>>>>>>>>>>>transform: '%s'>>>>>>>>>>", m_name.str());
+        ForEachItemIn(i, m_operations)
+            m_operations.item(i).toDBGLog();
+        DBGLOG("<<<<<<<<<<<<<<<<transform<<<<<<<<<<<<");
+#endif
+     }
+
+    virtual ~CEsdlCustomTransform(){}
+
+    void processTransform(IEspContext * context, IPropertyTree *theroot, IXpathContext *xpathContext, const char *target) override
```

Comment:
The naming of these three overloaded methods is confusing. This one is called by processServiceAndMethodTransforms() whereas the other two call processServiceAndMethodTransforms(). I think it would be easier to follow if one or the other sets had a different name.

Reply:
Wholeheartedly agree.  but it's been like that for a long time and I won't refactor that bit in this PR.  Maybe if I do a PR for fixing preservation of order.

Reply:
Changed my mind, I'll fix it now, it's very confusing.